rmw_zenoh_cpp/src/rmw_event.cpp

```diff
@@ -69,12 +69,14 @@ rmw_publisher_event_init(
   rmw_event->event_type = event_type;
 
   // Register the event with graph cache.
+  std::weak_ptr<rmw_zenoh_cpp::PublisherData> data_wp = pub_data;
```

Comment:
Is this a separate bugfix?

Reply:
The approach with capturing the `weak_ptr` ensures that if the event callback executes for some reason after the last reference to `PublisherData::SharedPtr` goes out of scope, we would safely exit the callback. This is what I want to do with some of the Zenoh callbacks as well once we switch to `zenoh-cpp`.
I would have liked to also do it within `rmw_subscription_event_init` but given that we're storing the `raw_ptr` to `SubscriptionData` in `rmw_subscription_t->data` and hence not able to retrieve the shared_ptr to the same, we are stuck with potential to dereference a dangling pointer if the scenario above. But this scenario should never occur given the implementation of `rcl*`